ircd/s_user.c

```diff
@@ -594,7 +594,11 @@ register_local_user(struct Client *client_p, struct Client *source_p)
 			SetDynSpoof(source_p);
 	}
 
-	source_p->umodes |= ConfigFileEntry.default_umodes & ~ConfigFileEntry.oper_only_umodes & ~orphaned_umodes;
+	source_p->umodes
+	  |= ( (ConfigFileEntry.default_umodes & ~aconf->umodes_mask)
+	     | (aconf->umodes                  &  aconf->umodes_mask))
+          & ~ConfigFileEntry.oper_only_umodes
+	  & ~orphaned_umodes;
```

Comment:
Maybe just split this out into several separate operations?  Would probably be easier to read that way.

```suggestion
	source_p->umodes |= ConfigFileEntry.default_umodes & ~aconf->umodes_mask;
	source_p->umodes |= aconf->umodes;
	source_p->umodes &= ~ConfigFileEntry.oper_only_umodes
	source_p->umodes &= ~orphaned_umodes;
```

(`aconf->umodes_mask` is redundant when used with `aconf->umodes`)

Reply:
```c
	umodes = ConfigFileEntry.default_umodes & ~aconf->umodes_mask;
	umodes |= aconf->umodes;
	umodes &= ~ConfigFileEntry.oper_only_umodes;
	umodes &= ~orphaned_umodes;
	source_p->umodes = umodes;
```